tests/visualization_tests/test_timeline.py

```diff
@@ -58,9 +66,16 @@ def test_get_timeline_info_empty() -> None:
     assert len(info.bars) == 0
 
 
-def test_get_timeline_info() -> None:
+@pytest.mark.parametrize(
+    "sys_attrs",
+    [
+        None,
+        {_CONSTRAINTS_KEY: [1.0]},
+    ],
+)
+def test_get_timeline_info(sys_attrs: Optional[Dict[str, Any]]) -> None:
```

Comment:
I noticed that the current test case does not cover the scenario where all constraint violations are negative. Could you verify it, for example, as follows?

```suggestion
@pytest.mark.parametrize(
    "trial_sys_attrs, infeasible",
    [
        (None, False),
        ({_CONSTRAINTS_KEY: [1.0]}, True),
        ({_CONSTRAINTS_KEY: [-1.0]}, False),
    ],
)
def test_get_timeline_info(trial_sys_attrs: Optional[Dict[str, Any]], infeasible: bool) -> None:
```

In addition, `sys_attrs` should be `trial_sys_attrs` to align with the argument of `_create_study`.